discovery-utils/proto/discovery.proto

```diff
@@ -0,0 +1,80 @@
+syntax = "proto3";
+
+package v0;
+
+// Registration is the service advertised by the Akri Agent.
+// Any `DiscoveryHandler` can register with Akri under a specified
+// protocol.
+service Registration {
+    rpc Register(RegisterRequest) returns (Empty) {}
+}
+
+
+message RegisterRequest {
+    // Name of the protocol that the client uses for discovery
+    string protocol = 1;
+    // Name of the endpoint that is hosting the discovery service
+    // for the protocol
+    string endpoint = 2;
+    // Specifies whether this device can only be ever seen by this node (e.g. a local USB device) 
+    // rather than being visible to multiple nodes (e.g. an IP camera) 
+    bool is_local = 3;
+}
+
+message Empty {
+}
+
+service Discovery {
```

Comment:
A point to ponder: In the future, we may consider exposing 'PreStartContainer' as well since we abstracting the Kubernetes implementation.

Reply:
What is `PreStartContainer` used for?